File: frontend/bhau_fitness_flutter/lib/models/workout_log.dart
/// Mirrors WorkoutLogDto on the backend.
class WorkoutLog {
  final int id;
  final String exercise;
  final int sets;
  final int reps;
  final double weightKg;
  final DateTime loggedDate;

  WorkoutLog({
    required this.id,
    required this.exercise,
    required this.sets,
    required this.reps,
    required this.weightKg,
    required this.loggedDate,
  });

  factory WorkoutLog.fromJson(Map<String, dynamic> json) {
    return WorkoutLog(
      id: json['id'] as int,
      exercise: json['exercise'] as String,
      sets: json['sets'] as int,
      reps: json['reps'] as int,
      weightKg: (json['weightKg'] as num).toDouble(),
      loggedDate: DateTime.parse(json['loggedDate'] as String),
    );
  }
}

File: backend/BhauFitnessApi/Models/Entities/WorkoutLog.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BhauFitnessApi.Models.Entities;

/// <summary>One logged exercise set, mirrors the HTML member portal's workout log.</summary>
public class WorkoutLog
{
    public int Id { get; set; }

    [Required]
    public string UserId { get; set; } = string.Empty;
    [ForeignKey(nameof(UserId))]
    public ApplicationUser? User { get; set; }

    [Required, MaxLength(120)]
    public string Exercise { get; set; } = string.Empty;

    public int Sets { get; set; }
    public int Reps { get; set; }

    [Column(TypeName = "decimal(6,2)")]
    public decimal WeightKg { get; set; }

    public DateOnly LoggedDate { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}
